modules/test/playwright/tests/layout-set-prototype-web/siteTemplatesPropagation.spec.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {expect, mergeTests} from '@playwright/test';
+
+import {applicationsMenuPageTest} from '../../fixtures/applicationsMenuPageTest';
+import {dataApiHelpersTest} from '../../fixtures/dataApiHelpersTest';
+import {featureFlagsTest} from '../../fixtures/featureFlagsTest';
+import {loginTest} from '../../fixtures/loginTest';
+import {pageEditorPagesTest} from '../../fixtures/pageEditorPagesTest';
+import {pageTemplatesPagesTest} from '../../fixtures/pageTemplatesPagesTest';
+import {pageViewModePagesTest} from '../../fixtures/pageViewModePagesTest';
+import {pagesAdminPagesTest} from '../../fixtures/pagesAdminPagesTest';
+import {productMenuPageTest} from '../../fixtures/productMenuPageTest';
+import {sitesPageTest} from '../../fixtures/sitesPageTest';
+import getRandomString from '../../utils/getRandomString';
+import createSiteTemplate from './utils/createSiteTemplate';
+
+export const test = mergeTests(
+	dataApiHelpersTest,
+	applicationsMenuPageTest,
+	loginTest(),
+	featureFlagsTest({
+		'LPD-39304': {enabled: true},
+	}),
+	pagesAdminPagesTest,
+	pageEditorPagesTest,
+	pageTemplatesPagesTest,
+	productMenuPageTest,
+	sitesPageTest,
+	pageViewModePagesTest
+);
+
+test('User is able to propagate pages separately on site templates', async ({
+	apiHelpers,
+	applicationsMenuPage,
+	page,
+	pageEditorPage,
+	pagesAdminPage,
+	productMenuPage,
+	sitesPage,
+	widgetPagePage,
+}) => {
+	const siteTemplateName: string = 'timplate-' + getRandomString();
+	const siteName: string = 'site-' + getRandomString();
+	const pageName: string = 'Page-' + getRandomString();
```

Comment:
I wonder if it's necessary to alter randomStrings with an additional prefix. If you see the need maybe we can unify the pattern (all starting with upper- or lowercase). Also, I believe you wanted to say template? 😅